internal/service/odb/db_server_data_source.go

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package odb
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/service/odb"
+	odbtypes "github.com/aws/aws-sdk-go-v2/service/odb/types"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-provider-aws/internal/create"
+	"github.com/hashicorp/terraform-provider-aws/internal/framework"
+	"github.com/hashicorp/terraform-provider-aws/internal/framework/flex"
+	fwtypes "github.com/hashicorp/terraform-provider-aws/internal/framework/types"
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+// Function annotations are used for datasource registration to the Provider. DO NOT EDIT.
+// @FrameworkDataSource("aws_odb_db_server", name="Db Server")
+func newDataSourceDbServer(context.Context) (datasource.DataSourceWithConfigure, error) {
+	return &dataSourceDbServer{}, nil
+}
+
+const (
+	DSNameDbServer = "Db Server Data Source"
+)
+
+type dataSourceDbServer struct {
+	framework.DataSourceWithModel[dbServerDataSourceModel]
+}
+
+func (d *dataSourceDbServer) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			names.AttrID: schema.StringAttribute{
+				Description: "The identifier of the the database server.",
+				Required:    true,
+			},
+			"cloud_exadata_infrastructure_id": schema.StringAttribute{
+				Description: "The identifier of the database server to retrieve information about.",
+				Required:    true,
+			},
+			"status": schema.StringAttribute{
+				CustomType:  fwtypes.StringEnumType[odbtypes.ResourceStatus](),
+				Computed:    true,
+				Description: "The status of the database server.",
+			},
+			"status_reason": schema.StringAttribute{
+				Computed:    true,
+				Description: "Additional information about the current status of the database server.",
+			},
+			"cpu_core_count": schema.Int32Attribute{
+				Computed:    true,
+				Description: "The number of CPU cores enabled on the database server.",
+			},
+			"db_node_ids": schema.ListAttribute{
+				Computed:    true,
+				CustomType:  fwtypes.ListOfStringType,
+				ElementType: types.StringType,
+			},
+			"db_node_storage_size_in_gbs": schema.Int32Attribute{
+				Computed:    true,
+				Description: "The allocated local node storage in GBs on the database server.",
+			},
+			"db_server_patching_details": schema.ObjectAttribute{
```

Comment:
we should set this as a `ListAttribute` with the associated custom type. For consistency with how we structure resources, we also avoid `ObjectAttributes` in datasources